sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/UnusedReturnValue.CSharp9.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+
+int MyMethod() { return 42; } // Compliant FN
+int MyMethod2() { return 42; }
+
+MyMethod();
+var i = MyMethod2();
+Action<int> a = (x) => MyMethod();
+SomeGenericMethod<object>();
+var o = SomeGenericMethod2<object>();
+
+T SomeGenericMethod2<T>() where T : class { return null; }
+T SomeGenericMethod<T>() where T : class { return null; } // Compliant
+
+new R().MyMethod();
+i = new R().MyMethod2();
+
+int NeverUsed(int neverUsedVal) // FN - the returned value is not used
```

Comment:
does this work for local functions in C# 7?

Reply:
Yes, we have a few test cases with local functions and static local functions and the analysis works as expected. It seems we have issues only with local functions defined as top level statements.